R/derive_vars_crit.R

```diff
@@ -0,0 +1,91 @@
+#' Derive Analysis Criterion Evaluation variables
+#'
+#' ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#' @description                                                                   +
+#' Derive analysis criterion evaluation result variable, paired with character    +
+#' and numeric flags.                                                             +
+#' ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#'
+#' @param dataset
+#' Input dataset
+#'
+#' @param new_var
+#' Variables to add
+#' The analysis criterion evaluation variable's name (i.e., CRIT1)
+#' This name is also used in order to create both charcater and numeric
+#' flags variables (i.e., CRIT1FL and CRIT1FN).
+#' Is possible to give a different name, base on needs (i.e., ANL01)
+#'
+#' @param label_var
+#' Criterion value
+#' A text description defining the condition necessary to satisfy the presence
+#' of the criterion
+#'
+#' @param condition
+#' Condition for selecting a subset
+#' The condition specified in order to select a subset from the input dataset
+#' in which the rule is applied.
+#'
+#' @param criterion
+#' Criterion rule
+#' The criterion that each selected row satisfies or not.
+#' Returns Y or N for character variable and 1 or 0 fro numeric variable
+#' if the criterion is met or not, respectively.
+#' Returns NA for not selected rows (not taken into account from condition)
+#'
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#' @return dataset with criterion variables
+#'
+#' @export
+#'
+#' @author Federico Baratin
+#'
+#' @examples
+#'
+#' is2 <- derive_vars_crit(
+#'   dataset = is,
+#'   new_var = "CRIT1",
+#'   label_var = "Titer >= ISLLOQ",
+#'   condition = !is.na(AVAL) & !is.na(ISLLOQ),
+#'   criterion = AVAL >= ISLLOQ
```

Comment:
where is it getting the is data from - the package isn't smart enough to find data?  You have to supply it yourself - you can make a dummy example using `tibble::tribble()` or call `data("is")` 

Check out `derive_locf_records()` for example.  

https://github.com/pharmaverse/admiral/blob/main/R/derive_locf_records.R

![image](https://user-images.githubusercontent.com/10111024/226205565-ce793586-fec2-45b3-a6f2-b9e63bce0522.png)

Reply:
Yeah, you're right.
Updated code with tribble example.
Also added family, keywords and export.